topostats/io.py

```diff
@@ -60,20 +65,101 @@ def write_yaml(config: dict, output_dir: Union[str, Path]) -> None:
             LOGGER.error(exception)
 
 
-def load_scan(img_path: Union[str, Path]) -> Bruker:
-    """Load the image from file.
+class LoadScan:
+    """Load the image and image parameters from a file path."""
 
-    Parameters
-    ----------
-    img_path : Union[str, Path]
-        Path to image that needs loading.
+    def __init__(
+        self,
+        img_path: Union[str, Path],
+        channel: str,
+    ):
 
-    Returns
-    -------
-    Bruker
+        """Initialise the class.
 
-    Examples
-    --------
-    """
-    LOGGER.info(f"Loading image from : {img_path}")
-    return Bruker(Path(img_path))
+        Parameters
+        ----------
+        img_path: Union[str, Path]
+            Path to a valid AFM scan to load.
+        channel: str
+            Image channel to extract from the scan.
+        """
+        self.img_path = Path(img_path)
+        self.channel = channel
+        self.channel_data = None
+        self.file_path = self.img_path
+        self.filename = self.img_path.stem
+        self.suffix = self.img_path.suffix
+        self.image = None
+        self.pixel_to_nm_scaling = None
+
+    def load_spm(self) -> tuple:
+        """Extract image and pixel to nm scaling from the Bruker .spm file."""
+        LOGGER.info(f"Loading image from : {self.img_path}")
+        try:
+            scan = Bruker(self.img_path)
+            LOGGER.info(f"[{self.filename}] : Loaded image from : {self.img_path}")
+            self.channel_data = scan.get_channel(self.channel)
+            LOGGER.info(f"[{self.filename}] : Extracted channel {self.channel}")
+            image = np.flipud(np.array(self.channel_data.pixels))
+        except FileNotFoundError:
+            LOGGER.info(f"[{self.filename}] File not found : {self.img_path}")
+        except Exception as exception:
+            LOGGER.error(f"[{self.filename}] : {exception}")
+
+        return (image, self._spm_pixel_to_nm_scaling(self.channel_data))
+
+    def _spm_pixel_to_nm_scaling(self, channel_data) -> float:
+        """Extract pixel to nm scaling from the SPM image metadata.
+
+        Parameters
+        ----------
+        channel_data:
+            Channel data
+
+        Returns
+        -------
+        float
+            Pixel to nm scaling factor.
+        """
+        unit_dict = {
+            "nm": 1,
+            "um": 1e3,
+        }
+        px_to_real = channel_data.pxs()
+        # Why are both nm and um calculated here and then only nm used?
```

Comment:
To answer this, it doesn't (or at least it's not meant to (its been a while since I've looked at this)).

`px_to_real` I think returns `{x_coord: [value, unit], y_coord: [value, unit]`. So, the list that is returned for this is one of the x_coord `px_to_nm_scaling` and the y_coord `px_to_nm_scaling`. 

Usually these are the same, and this is what I've seen with the test data but with the high speed afm, there may be a difference so this is here as I thought it may be useful in the future because if there were a difference we could more easily propagate this through.